backend/src/main/java/com/redhat/cloud/notifications/routers/NotificationResource.java

```diff
@@ -477,4 +482,44 @@ public List<BehaviorGroup> findBehaviorGroupsByBundleId(@Context SecurityContext
         );
         return behaviorGroups;
     }
+
+
+    @GET
+    @Path("/eventTypes/{eventTypeId}/endpoints")
+    @Produces(APPLICATION_JSON)
+    @Operation(summary = "Retrieve the endpoints linked to an event type.")
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_READ_NOTIFICATIONS)
+    public Page<Endpoint> getLinkedEndpoints(@Context SecurityContext sec, @PathParam("eventTypeId") UUID eventTypeId, @BeanParam @Valid Query query, @Context UriInfo uriInfo) {
+        String orgId = getOrgId(sec);
+
+        final List<Endpoint> endpoints = endpointEventTypeRepository.findEndpointsByEventTypeId(orgId, eventTypeId, query);
+
+        return new Page<>(
+            endpoints,
+            PageLinksBuilder.build(uriInfo.getPath(), endpoints.size(), query),
+            new Meta(Long.valueOf(endpoints.size()))
+        );
+    }
+
+    @DELETE
+    @Path("/eventTypes/{eventTypeId}/endpoint/{endpointId}")
+    @Operation(summary = "Delete the link between an endpoint and an event type", description = "Delete the link between an endpoint and an event type.")
+    @RolesAllowed(ConsoleIdentityProvider.RBAC_WRITE_NOTIFICATIONS)
+    @APIResponse(responseCode = "204")
+    public void deleteEndpointFromEventType(@Context final SecurityContext securityContext, @RestPath final UUID eventTypeId, @RestPath final UUID endpointId) {
+        final String orgId = getOrgId(securityContext);
+
+        endpointEventTypeRepository.deleteEndpointFromEventType(eventTypeId, endpointId, orgId);
+    }
```

Comment:
Same here, I think we need to check if both the specified event type and the endpoint exist in our end, and if not, return a 404.

Reply:
Sure